File: backend/src/FitnessTracker.Core/Entities/ExerciseType.cs
namespace FitnessTracker.Core.Entities;

public class ExerciseType
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public decimal DefaultMET { get; set; } = 5.0m;
    public bool IsSystemDefault { get; set; }
    public Guid? CreatedByUserId { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    public bool IsDeleted { get; set; }

    // Navigation properties
    public ICollection<Equipment> Equipments { get; set; } = new List<Equipment>();
    public ICollection<WorkoutRecord> WorkoutRecords { get; set; } = new List<WorkoutRecord>();
}

File: frontend/src/types/exercise-type.ts
export interface Equipment {
  id: number
  name: string
  description?: string
  isSystemDefault: boolean
  createdAt: string
  isDisabled: boolean
}

export interface ExerciseType {
  id: number
  name: string
  description?: string
  isSystemDefault: boolean
  defaultMET: number
  createdAt: string
  equipments: Equipment[]
  workoutRecordCount: number
  isDisabled: boolean
}

export interface CreateExerciseTypeDto {
  name: string
  description?: string
  defaultMET: number
  equipmentIds: number[]
}

export interface UpdateExerciseTypeDto {
  name: string
  description?: string
  defaultMET: number
  equipmentIds: number[]
}

export interface CreateEquipmentDto {
  name: string
  description?: string
}

export interface UpdateEquipmentDto {
  name: string
  description?: string
}
